src/views/Generator/Generator.hooks.ts

```diff
@@ -89,3 +90,39 @@ export function useIsGeneratorDirty(fileName: string) {
     JSON.stringify(generatorStateData) !== JSON.stringify(generatorFileData)
   )
 }
+
+export function useScriptExport(generatorFileName: string) {
```

Comment:
This hook is used in `GeneratorControls` and `ScriptPreview` to keep the behavior consistent.